### ISSUE TITLE
Poking the palette index doesn't affect `circ()` anymore

### ISSUE THREAD

**Issue by borbware** [nesbox/TIC-80#1960]
I noticed that in 1.0, my palette cycling functions don't work anymore for `circ` calls as they did in 0.9. `rect` in comparison works fine.

0.90.1723:
![video2](https://user-images.githubusercontent.com/19384640/173785762-3d1acc0c-5742-4bfd-a061-5ae980c77533.gif)

1.0.2164:
![video3](https://user-images.githubusercontent.com/19384640/173785786-1d1650ec-dfa3-4cfa-a955-dae07c0e370d.gif)

Here's the code to replicate the issue.
```Lua
function pal(c0,c1)--replace c0 with c1
	if(c0==nil and c1==nil)then for i=0,15 do poke4(0x3FF0*2+i,i)end
	else poke4(0x3FF0*2+c0,c1) end
end

function palcycl(T,cols,cols2)--cycle palette
	T=T or 0
	if cols and cols2 then
		for i,v in ipairs(cols)do
			local col2=cols2[(i-1+T//1)%#cols2+1]
			pal(v,col2)
		end
	else
		for i=0,15 do pal(i,(i+T)%16)end
	end
	return retcol
end

t=0
x=120
y=60
palette={{1,2,6,8,10,14,15},{1,2,6,8,10,14,15}}
function TIC()
	cls(13)
	palcycl(t//8,palette[1],palette[2])
	circ(x,y,10,6)
	rect(x,y+20,10,10,6)
	palcycl()
	t=t+1
end
```

**Comment by potato25** [comment]
i actually prefer circ with 0 radius drawing a single pixel , lot of my carts work that way

**Comment by nesbox** [comment]
> i actually prefer circ with 0 radius drawing a single pixel , lot of my carts work that way

I don't know how to please everyone, one says he doesn't want to see anything with zero radius, the other wants to see a 1px dot :)

**Comment by joshgoebel** [comment]
> Also, a circle with zero radius shouldn't draw anything (now draws a 1px dot).

@nesbox Why would this be bad?  I mean _technically_ a circle with a radius 0 would be a point, not a circle, but what is the harm if it renders as a point?  If someone doesn't want to draw anything they shouldn't call a drawing function... I'm not sure there is a 100% right or wrong here but I don't see any problem with a 0px radius circle drawing a point/pixel.

**Comment by borbware** [comment]
i'm fine either way. Maybe leaning towards 0 radius drawing 1px dot, **only** because that's how it has worked previously, and changing the behaviour would change how some old cartridges look.

**Comment by joshgoebel** [comment]
Closing this as resolved by #1960.  If we want to continue the 0 vs 1 pixel radius discussion we should open a separate issue.

**Comment by borbware** [comment]
yeah good call as if was fixed in d23ffe1daebf33780f6b8f7bc6d7c83c2716af0c.
